isis/src/cassini/apps/ciss2isis/ciss2isis.cpp

```diff
@@ -0,0 +1,449 @@
+    pixel.push_back(swapper.ShortInt(& (data[22])));
+    pixel.push_back(ComputeOverclockAvg(pixel));
+    for(int i = 0; i < (int)pixel.size(); i++) {
+      pixelBuf[0] = pixel[i];
+      //  Do 8 bit to 12 bit conversion for prefix data
+      FixDns(pixelBuf);
+      double pix = pixelBuf[0];
+      if(pix == NULL8) {
+        calibrationPixels.push_back(NULL2);
+      }
+      else if(pix == LOW_REPR_SAT8) {
+        calibrationPixels.push_back(LOW_REPR_SAT2);
+      }
+      else if(pix == LOW_INSTR_SAT8) {
+        calibrationPixels.push_back(LOW_INSTR_SAT2);
+      }
+      else if(pix == HIGH_INSTR_SAT8) {
+        calibrationPixels.push_back(HIGH_INSTR_SAT2);
+      }
+      else if(pix == HIGH_REPR_SAT8) {
+        calibrationPixels.push_back(HIGH_REPR_SAT2);
+      }
+      else {
+        calibrationPixels.push_back(pix);
+      }
+    }
+    return calibrationPixels;
+  }
+
+  // Called in IsisMain() if DataConversionType == Table
+  // Creates stretch pairs for mapping in Fix Dns
+  // author Jeannie Walldren 2008-08-21
+  void CreateStretchPairs() {
+    // Set up the strech for the 8 to 12 bit conversion from file
+    PvlGroup &dataDir = Preference::Preferences().findGroup("DataDirectory");
+    QString missionDir = (QString) dataDir["Cassini"];
+    FileName *lutFile = new FileName(missionDir + "/calibration/lut/lut.tab");
+    CisscalFile *stretchPairs = new CisscalFile(lutFile->expanded());
+    // Create the stretch pairs
+    double temp1 = 0;
+    stretch.ClearPairs();
+    for(int i = 0; i < stretchPairs->LineCount(); i++) {
+      QString line;
+      stretchPairs->GetLine(line);  //assigns value to line
+      line = line.simplified();
+
+      foreach (QString value, line.split(QRegExp("[\\s,]"), QString::SkipEmptyParts)) {
+        stretch.AddPair(temp1, toDouble(value));
+        temp1++;
+      }
+    }
+    stretchPairs->Close();
+    return;
+  }
+
+  /**
+  * The input buffer has a raw 16 bit buffer but the values are still 0 to 255.
+  * We know that 255 (stretched to 4095 if Table converted) is saturated.
+  * Sky pixels could have valid DN of 0, but missing pixels are also saved as 0,
+  * so it is impossible to distinguish between them.
+  * This method is used by ConvertLinePrefixPixels() and IsisMain() for ProcessByLine p2.
+  * author Jeannie Walldren 2008-08-21
+  *
+  */
+  void FixDns(Buffer &buf) {
+    for(int i = 0; i < buf.size(); i++) {
+      // zeros and negatives are valid DN values, according to scientists,
+      // but likelyhood of a zero in 16 bit is rare,
+      // so assume these are missing pixels and set them to null
+      if(buf[i] == 0) {
+        buf[i] = Null;
+      }
+      else if(dataConversionType == "Table") {
+        buf[i] = stretch.Map((int)buf[i]);
+      }
+      // save max values (4095 for table-converted images and 255 for others) as HRS
+      if(buf[i] >= validMax) {
+        buf[i] = Hrs;
+      }
+    }
+  }
```

Comment:
@jessemapel This function was modified on line 242 changing from `buf[i] == validMax` to `buf[i] >= validMax`. As I mentioned, @scsides and I discussed how valid it would be to actually have `buf[i] > validMax` which is what the variable validMax seems to imply. However, reading the comment on line 241 seems to imply differently

Reply:
From the SIS, it seems like this value indicates the maximum possible DN. So, it would be ambiguous if the true amount of exposure for that pixel was right at that number or above it.

https://github.com/USGS-Astrogeology/ISIS3/issues/672#issuecomment-448812918

I think >= is correct here.

Reply:
That's that way we saw it.